### PR TITLE
dts: stm32: fix adc1 reg and clocks properties

### PR DESCRIPTION
**Pull request by passgat**
Set the correct address space and the bus settings to enable its clock.

Signed-off-by: Dario Binacchi <dariobin@libero.it>

### CODE REVIEW

File: dts/arm/st/f3/stm32f3.dtsi [hunk 1]
```diff
@@ -348,10 +348,10 @@
 			phase-seg2 = <6>;
 		};
 
-		adc1: adc@50000000 {
+		adc1: adc@40012400 {
 			compatible = "st,stm32-adc";
-			reg = <0x50000000 0x400>;
-			clocks = <&rcc STM32_CLOCK_BUS_AHB1 0x10000000>;
+			reg = <0x40012400 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB2 0x00000200>;
```
Comment (erwango) [inline dts/arm/st/f3/stm32f3.dtsi:354]: Actually, the initial code is (partially) right:
ADC1 is located at @5000000 for most variants of STM32F3 series. What's wrong is the bus identifier, should be STM32_CLOCK_BUS_AHB3.
If this address is valid for you, I guess you're using either stm32f373 or stm32f378 variant.

In that case, I'd advice the following:
Remove adc1 from stm32f3.dtsi and add it directly, with respective correct values in each of the SoC variant files: (stm32f302.dtsi, stm32f303.dtsi, ...)

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by passgat** [comment]
Hi Erwan,


>     Il 22/03/2021 09:36 Erwan Gouriou ***@***.***> ha scritto:
> 
> 
> 
> 
>     @erwango requested changes on this pull request.
> 
> 
>     ---------------------------------------------
> 
>     In dts/arm/st/f3/stm32f3.dtsi https://github.com/zephyrproject-rtos/zephyr/pull/33534#discussion_r598512398 :
> 
>     > -  adc1: ***@***.*** {
>     +  adc1: ***@***.*** {
>         compatible = "st,stm32-adc";
>     -   reg = <0x50000000 0x400>;
>     -   clocks = <&rcc STM32_CLOCK_BUS_AHB1 0x10000000>;
>     +   reg = <0x40012400 0x400>;
>     +   clocks = <&rcc STM32_CLOCK_BUS_APB2 0x00000200>;
> 
>     Actually, the initial code is (partially) right:
>     ADC1 is located at @5000000 https://github.com/5000000 for most variants of STM32F3 series. What's wrong is the bus identifier, should be STM32_CLOCK_BUS_AHB3.
>     If this address is valid for you, I guess you're using either stm32f373 or stm32f378 variant.
> 

You are right, I'am using stm32f373.

> 
>     In that case, I'd advice the following:Remove adc1 from stm32f3.dtsi and add it directly, with respective correct values in each of the SoC variant files: (stm32f302.dtsi, stm32f303.dtsi, ...)
> 

I'll follow your suggestions and I'll change the patch 

Thanks and regards,
Dario

**Comment by erwango** [comment]
> If in the reference manual of the stm32f373 I find the RCC_APB2ENR (APB2 bus) register, in the manuals of the other microcontrollers,
> in the section where the address space is exposed, reference is made to the AHB3 bus, but I only find the AHB register description (
> where i found a bit for enabling the ADC1). I would have expected a register for the AHB3 bus.

Good catch.
So indeed, in the whole F3 series, there is only one AHB bus.
In CMSIS SoC descriptions files, we can see references to AHB2 and AHB3, but this is only virtual sub bus groups.
In `stm32cube/stm32f3xx/drivers/include/stm32f3xx_ll_bus.h`, all AHBX peripherals clocks should actually be controlled 
using `LL_AHB1_GRP1_EnableClock` function.
As a consequence, for the ADC devices referenced as being on AHB3:
to fit Zephyr clock control driver, they should be described in device tree as on `STM32_CLOCK_BUS_AHB1`.

**Comment by passgat** [comment]
Hi Erwan, 
I know you are busy, but have you found time to look at my patch?
I have uploaded the changes you requested.
It is my first pull request for the Zephyr project and I doubt that I have done 
something wrong with the patch submission workflow.

Thanks and regards
Dario

**Review by erwango — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**